.github/workflows/release-apt-get.yml

```diff
@@ -0,0 +1,52 @@
+name: "release-apt-get"
+on:
+  release:
+    types: [released]
+
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release Tag'
+        required: true
+        default: 'latest'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - name: setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - uses: actions/checkout@v2
+
+    - uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: 'Download Repo Client'
+      env:
+        AZ_SUB: ${{ secrets.AZURE_SUBSCRIPTION }}
+      run: |
+        az storage blob download --subscription  "$AZ_SUB" --account-name gitcitoolstore -c tools -n azure-repoapi-client_2.0.1_amd64.deb -f repoclient.deb --auth-mode login
+    - name: "Install Repo Client"
+      env:
+        AZURE_AAD_ID: ${{ secrets.AZURE_AAD_ID }}
+        AAD_CLIENT_SECRET: ${{ secrets.AAD_CLIENT_SECRET }}
+      run: |
+        sudo apt-get install python3-adal --yes
+        sudo dpkg -i repoclient.deb
+        python .github/configure_repoclient.py
+        rm repoclient.deb
+    - name: "Publish to apt feed"
+      env:
+        RELEASE: ${{ github.event.inputs.release }}
+      run: |
+        pip install requests
+        python .github/fetch_release.py
```

Comment:
It might make sense to turn this into a [`github-script`](https://github.com/marketplace/actions/github-script) task instead: that allows you to use the REST API without much hassle and with less code. You can probably copy-edit what I did [in  `setup-git-for-windows-sdk`](https://github.com/git-for-windows/setup-git-for-windows-sdk/blob/e377dda38b797976b20f6dd45098f306d5b0b137/.github/workflows/release-tag.yml#L64-L81).

Reply:
Sure, I'm fine with that. I'm just having a little trouble figuring out what [this piece](https://github.com/git-for-windows/setup-git-for-windows-sdk/blob/e377dda38b797976b20f6dd45098f306d5b0b137/.github/workflows/release-tag.yml#L73-L81) is using. Is there a corresponding method for getting a release (I assume so, but want to be sure)? Would you mind pointing me to any existing docs? I've done a couple searches but am having some trouble getting a matching result.

Reply:
I pushed an example demonstrating this idea to my fork: https://github.com/dscho/git/actions/runs/966065373/workflow

Reply:
The essence of this example is this step:

```yml
      - name: "Get release assets"
        id: get-asset
        env:
          RELEASE: ${{ github.event.inputs.release }}
        uses: actions/github-script@v3
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            const { data } = await github.repos.getRelease({
                owner: 'microsoft', // context.repo.owner,
                repo: context.repo.repo,
                release_id: process.env.RELEASE || 'latest'
              })
            const assets = data.assets.filter(asset => asset.name.endsWith('.deb'))
            if (assets.length !== 1) {
              throw new Error(`Unexpected number of .deb assets: ${assets.length}`)
            }
            core.setOutput('name', assets[0].name)
            core.setOutput('url', assets[0].url)
```

In subsequent steps, you can use `${{ steps.get-asset.outputs.name }}` and `${{ steps.get-asset.outputs.url }}` to refer to the discovered asset name/URL.

BTW I could imagine that things would look a bit simpler if you used another `github-script` step to generate and write the JSON configuration. Something along these lines:

```yml
      - name: "Write repoclient config"
        env:
          AZURE_AAD_ID: ${{ secrets.AZURE_AAD_ID }}
          AAD_CLIENT_SECRET: ${{ secrets.AAD_CLIENT_SECRET }}
        uses: actions/github-script@v3
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            for (const key of ['AZURE_AAD_ID', 'AAD_CLIENT_SECRET']) {
              if (!process.env[key]) throw new Error(`Required env var ${key} is missing!`)
            }

            const config = {
              AADResource: 'https://microsoft.onmicrosoft.com/945999e9-da09-4b5b-878f-b66c414602c0',
              AADTenant: '72f988bf-86f1-41af-91ab-2d7cd011db47',
              AADAuthorityUrl: 'https://login.microsoftonline.com',
              server: 'azure-apt-cat.cloudapp.net',
              port: '443',

              AADClientId: process.env.AZURE_AAD_ID,
              AADClientSecret: process.env.AAD_CLIENT_SECRET,
              repositoryId: ''
            }

            const fs = require('fs')
            fs.writeFileSync('config.json', JSON.stringify(config, null, 2))
```

Technically, this is "abusing" the `github-script` Action a bit because we are not even attempting to access any GitHub resources. But practically it is a really convenient way to execute Javascript snippets.

Reply:
Thank you so much for the detailed description, @dscho! This will be invaluable to my GitHub Actions flows far into the future :).